sqlitecluster/SQLite.cpp

```diff
@@ -282,8 +284,26 @@ void SQLite::_sqliteLogCallback(void* pArg, int iErrCode, const char* zMsg) {
     // This is sort of hacky to parse this from the logging info. If it works we could ask sqlite for a better interface to get this info.
     if (SStartsWith(zMsg, "cannot commit")) {
         // 17 is the length of "conflict at page" and the following space.
-        const char* offset = strstr(zMsg, "conflict at page") + 17;
-        _conflictPage = atol(offset);
+        const char* pageOffset = strstr(zMsg, "conflict at page") + 17;
+        _conflictPage = atol(pageOffset);
+
+        // 17 is the length of "part of db table" and the following space.
+        const char* tableOffset = strstr(pageOffset, "part of db table") + 17;
+
+        // Check if the tableOffset exists since not all conflicts are on tables
+        if (tableOffset) {
+            // Based on the SQLite log line, we should always have ';' after the table name,
+            // so let's find it and use it to limit the size of the substring we need
+            const char* semicolonOffset = strstr(tableOffset, ";");
+            
+            // Let's add this check in case the SQLite log changes and we don't notice it
+            // since this would generate a runtime error.
+            if (semicolonOffset) {
```

Comment:
added this IF since the string initializer would generate a runtime error in case semicollongOffset was null: 
![image](https://github.com/user-attachments/assets/6dc96226-5603-42e9-9a60-442cd09f2c72)

Reply:
I don't understand this comment. You need the `if`, yes, because otherwise `semicolonOffset - tableOffset` will be negative.

The error you posted looks like you were setting `msg` by hand, and that's the error you're seeing.